lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -152,8 +172,14 @@ def self.bulk_destroy(elements)
         end
 
         PolicyElement.where(unique_identifier: elements.keys).delete_all
-        Assignment.where(parent_id: elements.values.flat_map(&:id)).delete_all
-        Assignment.where(child_id: elements.values.flat_map(&:id)).delete_all
+
+        ids = elements.values.flat_map(&:id)
+        Assignment.where(parent_id: ids).delete_all
```

Comment:
Maybe `[Assignment, CrossAssignment].each do` ? Or even `[Assignment, CrossAssignment].product([:parent_id, :child_id]) do` if you wanna golf.